src/copy.c

```diff
@@ -396,14 +396,41 @@ copyfrom(CopyChunkState *ccstate, List *range_table, Hypertable *ht, void (*call
 				resultRelInfo->ri_RelationDesc->rd_att->constr)
 			{
 				Assert(resultRelInfo->ri_RangeTableIndex > 0 && estate->es_range_table);
-				ExecConstraints(resultRelInfo, myslot, estate);
+				ExecConstraints(saved_resultRelInfo, myslot, estate);
+			}
+			else if (cis->compress_state &&
```

Comment:
Also would be good to add a comment on why this is necessary (as part of this PR or the follow-up).

Reply:
reverted not sure why it was there might have been testing something